baremetal/metal3machine_manager.go

```diff
@@ -1371,7 +1375,7 @@ func (m *MachineManager) SetNodeProviderID(ctx context.Context, bmhID *string, p
 	if matchingNodesCount == 1 {
 		return nil
 	}
-	if bmhID == nil {
+	if bmhID == "" {
```

Comment:
I'm not sure if we should keep this. The only reason it would happen is if the bmhID is explicitly set to `""` if I understand it correctly, and in that case we are stuck anyway. So I'm thinking it would be better to add a check in `getBmhIDFromM3Machine` to see if `host.UID == ""`, if it is, we return an error there directly. What do you think?

Reply:
Yes, I agree, I think I should also remove the `if bmhID != "" {` check on the line 1354.

Reply:
Good point, it makes sense to simplify that also! :slightly_smiling_face: